test/ck_tile/atomic_add_op/test_atomic.cpp

```diff
@@ -0,0 +1,103 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2025, Advanced Micro Devices, Inc. All rights reserved.
+
+#include <algorithm>
+#include <gtest/gtest.h>
+
+#include "ck_tile/host.hpp"
+#include "ck_tile/core.hpp"
+#include "ck_tile/host/kernel_launch.hpp"
+#include "test_atomic.hpp"
+
+struct AtomicKernelParam
+{
+    AtomicKernelParam(ck_tile::index_t m_, ck_tile::index_t n_) : m(m_), n(n_) {}
+    ck_tile::index_t m;
+    ck_tile::index_t n;
+};
+
+template <typename DataType_>
+class TestAtomicKernel : public ::testing::TestWithParam<std::tuple<int, int>>
+{
+    protected:
+    void RunTest(const AtomicKernelParam& params)
+    {
+        using XDataType = DataType_;
+
+        ck_tile::index_t m = params.m;
+        ck_tile::index_t n = params.n;
+
+        std::cout << "Input Tensor Dimensions: " << m << ", " << n << std::endl;
+
+        constexpr auto dword_bytes = 4;
+
+        if(n % (dword_bytes / sizeof(XDataType)) != 0)
+        {
+            std::cerr << "n size should be multiple of dword_bytes" << std::endl;
```

Comment:
if this condition is a deal breaker, then lets fail the test here

```suggestion
            ASSERT_EQ(n % (dword_bytes / sizeof(XDataType)), 0) << "n must be divisible by vector width";
```

Reply:
Done.